RoddyCore/src/de/dkfz/roddy/knowledge/brawlworkflows/BrawlWorkflow.groovy

```diff
@@ -1,417 +1,199 @@
+        return ExecutionService.getInstance().callSynchronized(context, toolID, parameters);
+    }
 
-            classToLoad = method.genericReturnType;
-        }
-        Field field = _classOfCallingObject.fields.find { Field f -> f.name == splitCall[indexInSplit].split("[(]")[0].replaceAll("[()]", "") }
-        if (field) {
-            classToLoad = field.genericType;
-        }
+    protected boolean getflag(String flagID, boolean defaultValue = true) {
+        return context.getConfiguration().getConfigurationValues().getBoolean(flagID, defaultValue);
+    }
 
-        if (splitCall.size() == indexInSplit + 1)
-            return classToLoad;
-        else {
-            //Is the next in line a method or a field?...
-            //If the current classToLoad is some sort of generic object... then what?
-            // Unfortunately, Roddy uses generics at some points, e.g. for Tuples. However, Generics can be a huge pain.
-            // If our next call is a method, we will just go on, if it is a value and the current type is generic... Let's handle that differently.
-            boolean nextIsMethod = isNextMethodCall(method ? method.returnType : field.type, splitCall, indexInSplit + 1)
-            if (nextIsMethod || (!nextIsMethod && !classToLoad.typeName.contains("<")))
-                return findTypeOfMethodOrProperty(method ? method.returnType : field.type, splitCall, indexInSplit + 1);
-            else {
-                // Now handle the generic fields...
-                //Get the fields name first.
-                String fieldName = splitCall[indexInSplit + 1];
-                int index = method.getReturnType().fields.findIndexOf { Field it -> it.name == fieldName; };
+    protected BaseFile getSourceFile(String path, String _class = BaseFile.STANDARD_FILE_CLASS) {
+        return BaseFile.fromStorage(context, path, _class);
+    }
 
-                Type genericReturnType = method.getGenericReturnType()
-                Field hidden = genericReturnType.class.getDeclaredField("actualTypeArguments");
-                hidden.setAccessible(true);
-                return ((Type[]) hidden.get(genericReturnType))[index];
-            }
-        }
+    protected BaseFile getSourceFileUsingTool(String toolID, String _class = BaseFile.STANDARD_FILE_CLASS)
+            throws ExecutionException, UnexpectedExecutionResultException {
+        return BaseFile.getSourceFileUsingTool(context, toolID, _class);
     }
 
-    /**
-     * Auto boolean to true or false.
-     * @param _classOfCallingObject
-     * @param splitCall
-     * @param indexInSplit
-     * @return
-     */
-    private static boolean isNextMethodCall(Class _classOfCallingObject, String[] splitCall, int indexInSplit) {
-        return _classOfCallingObject.methods.find { Method m -> m.name == splitCall[indexInSplit].replaceAll("[()]", "") }
+    protected List<BaseFile> getSourceFilesUsingTool(String toolID, String _class = BaseFile.STANDARD_FILE_CLASS)
```

Comment:
If we'd have dedicated ToolID, File, etc. types we could make this type save (instead of String) and we could shorten the DSL method names to a simple getSourceFile().

Maybe use declarator (run tool("toolA"), ...)?
```groovy
ToolID tool(String name) { new ToolID(name) }
```